Crashes when resuming if the panel is open when Activity is killed.

Buddybuild detected a crash

[View Full Crash Details](https://dashboard.buddybuild.com/apps/594b679a6ed87e0001fca2cd/crashreport/8f934fcafbb3de3f21c7e7b2bbc475ab)

Process: org.mozilla.zerda.debug.tyu, PID: 20827
                                                                             java.lang.RuntimeException: Unable to start activity ComponentInfo{org.mozilla.zerda.debug.tyu/org.mozilla.focus.activity.MainActivity}: java.lang.RuntimeException: Parcel android.os.Parcel@9159a00: Unmarshalling unknown type code 7471204 at offset 25052
                                                                                 at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2666)
                                                                                 at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2727)
                                                                                 at android.app.ActivityThread.-wrap12(ActivityThread.java)
                                                                                 at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1478)
                                                                                 at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                                 at android.os.Looper.loop(Looper.java:154)
                                                                                 at android.app.ActivityThread.main(ActivityThread.java:6121)
                                                                                 at java.lang.reflect.Method.invoke(Native Method)
                                                                                 at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
                                                                                 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
                                                                              Caused by: java.lang.RuntimeException: Parcel android.os.Parcel@9159a00: Unmarshalling unknown type code 7471204 at offset 25052
                                                                                 at android.os.Parcel.readValue(Parcel.java:2444)
                                                                                 at android.os.Parcel.readArrayMapInternal(Parcel.java:2732)
                                                                                 at android.os.BaseBundle.unparcel(BaseBundle.java:269)
                                                                                 at android.os.Bundle.getSparseParcelableArray(Bundle.java:934)
                                                                                 at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1208)
                                                                                 at android.support.v4.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1528)
                                                                                 at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1595)
                                                                                 at android.support.v4.app.FragmentManagerImpl.dispatchCreate(FragmentManager.java:2893)
                                                                                 at android.support.v4.app.FragmentController.dispatchCreate(FragmentController.java:190)
                                                                                 at android.support.v4.app.FragmentActivity.onCreate(FragmentActivity.java:353)
                                                                                 at android.support.v7.app.AppCompatActivity.onCreate(AppCompatActivity.java:85)
                                                                                 at org.mozilla.focus.locale.LocaleAwareAppCompatActivity.onCreate(LocaleAwareAppCompatActivity.java:39)
                                                                                 at org.mozilla.focus.activity.MainActivity.onCreate(MainActivity.java:59)
                                                                                 at android.app.Activity.performCreate(Activity.java:6682)
                                                                                 at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1118)
                                                                                 at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2619)
                                                                                 at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2727) 
                                                                                 at android.app.ActivityThread.-wrap12(ActivityThread.java) 
                                                                                 at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1478) 
                                                                                 at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                                                 at android.os.Looper.loop(Looper.java:154) 
                                                                                 at android.app.ActivityThread.main(ActivityThread.java:6121) 
                                                                                 at java.lang.reflect.Method.invoke(Native Method) 
                                                                                 at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889) 
                                                                                 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779) 

I might have reproduced this one:

STR: give and revoke download permission and come back to Zerda

It's still here

Process: org.mozilla.zerda.debug.tyu, PID: 24617
                                                   java.lang.RuntimeException: Unable to start activity ComponentInfo{org.mozilla.zerda.debug.tyu/org.mozilla.focus.activity.MainActivity}: java.lang.RuntimeException: Parcel android.os.Parcel@59a47ad: Unmarshalling unknown type code 7471204 at offset 15196
                                                       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2666)
                                                       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2727)
                                                       at android.app.ActivityThread.-wrap12(ActivityThread.java)
                                                       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1478)
                                                       at android.os.Handler.dispatchMessage(Handler.java:102)
                                                       at android.os.Looper.loop(Looper.java:154)
                                                       at android.app.ActivityThread.main(ActivityThread.java:6121)
                                                       at java.lang.reflect.Method.invoke(Native Method)
                                                       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
                                                    Caused by: java.lang.RuntimeException: Parcel android.os.Parcel@59a47ad: Unmarshalling unknown type code 7471204 at offset 15196
                                                       at android.os.Parcel.readValue(Parcel.java:2444)
                                                       at android.os.Parcel.readArrayMapInternal(Parcel.java:2732)
                                                       at android.os.BaseBundle.unparcel(BaseBundle.java:269)
                                                       at android.os.Bundle.getSparseParcelableArray(Bundle.java:934)
                                                       at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1208)
                                                       at android.support.v4.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1528)
                                                       at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1595)
                                                       at android.support.v4.app.FragmentManagerImpl.dispatchCreate(FragmentManager.java:2893)
                                                       at android.support.v4.app.FragmentController.dispatchCreate(FragmentController.java:190)
                                                       at android.support.v4.app.FragmentActivity.onCreate(FragmentActivity.java:353)
                                                       at android.support.v7.app.AppCompatActivity.onCreate(AppCompatActivity.java:85)
                                                       at org.mozilla.focus.locale.LocaleAwareAppCompatActivity.onCreate(LocaleAwareAppCompatActivity.java:39)
                                                       at org.mozilla.focus.activity.MainActivity.onCreate(MainActivity.java:61)
                                                       at android.app.Activity.performCreate(Activity.java:6682)
                                                       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1118)
                                                       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2619)
                                                       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2727) 
                                                       at android.app.ActivityThread.-wrap12(ActivityThread.java) 
                                                       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1478) 
                                                       at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                       at android.os.Looper.loop(Looper.java:154) 
                                                       at android.app.ActivityThread.main(ActivityThread.java:6121) 
                                                       at java.lang.reflect.Method.invoke(Native Method) 
                                                       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889) 
                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779) 
08-23 09:46:18.669 5977-5977/? E/OverlaySController: Calling with an invalid UID, the interface will not work
08-23 09:46:18.669 5977-5977/? E/OverlaySController: Invalid version
08-23 09:46:19.877 24090-24090/? E/ctxmgr: [PlaceFenceHelper]NearbyBuffer is null!
08-23 09:46:41.209 1178-1600/? E/ConnectivityService: RemoteException caught trying to send a callback msg for NetworkRequest [ LISTEN id=37, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED] ]
08-23 09:47:02.517 23655-24647/? E/Bugle: Get mms config failed: invalid subId. subId=-1, real subId=-1, map=[]

setTargetFragment() is discouraged
Ref: https://stackoverflow.com/questions/29714476/fragment-no-longer-exists-for-key-androidtarget-state, https://issuetracker.google.com/issues/36969568

                                                                         Process: org.mozilla.zerda.debug.tyu, PID: 18287
                                                                         java.lang.RuntimeException: Unable to start activity ComponentInfo{org.mozilla.zerda.debug.tyu/org.mozilla.focus.activity.MainActivity}: java.lang.IllegalStateException: Fragment no longer exists for key android:target_state: index 1
                                                                             at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2666)
                                                                             at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2727)
                                                                             at android.app.ActivityThread.-wrap12(ActivityThread.java)
                                                                             at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1478)
                                                                             at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                             at android.os.Looper.loop(Looper.java:154)
                                                                             at android.app.ActivityThread.main(ActivityThread.java:6121)
                                                                             at java.lang.reflect.Method.invoke(Native Method)
                                                                             at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
                                                                             at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
                                                                          Caused by: java.lang.IllegalStateException: Fragment no longer exists for key android:target_state: index 1
                                                                             at android.support.v4.app.FragmentManagerImpl.getFragment(FragmentManager.java:874)
                                                                             at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1210)
                                                                             at android.support.v4.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1528)
                                                                             at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1595)
                                                                             at android.support.v4.app.FragmentManagerImpl.dispatchCreate(FragmentManager.java:2893)
                                                                             at android.support.v4.app.Fragment.restoreChildFragmentState(Fragment.java:1294)
                                                                             at android.support.v4.app.Fragment.onCreate(Fragment.java:1265)
                                                                             at android.support.v4.app.DialogFragment.onCreate(DialogFragment.java:291)
                                                                             at android.support.v4.app.Fragment.performCreate(Fragment.java:2180)
                                                                             at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1244)
                                                                             at android.support.v4.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1528)
                                                                             at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1595)
                                                                             at android.support.v4.app.FragmentManagerImpl.dispatchCreate(FragmentManager.java:2893)
                                                                             at android.support.v4.app.FragmentController.dispatchCreate(FragmentController.java:190)
                                                                             at android.support.v4.app.FragmentActivity.onCreate(FragmentActivity.java:353)
                                                                             at android.support.v7.app.AppCompatActivity.onCreate(AppCompatActivity.java:85)
                                                                             at org.mozilla.focus.locale.LocaleAwareAppCompatActivity.onCreate(LocaleAwareAppCompatActivity.java:39)
                                                                             at org.mozilla.focus.activity.MainActivity.onCreate(MainActivity.java:81)
                                                                             at android.app.Activity.performCreate(Activity.java:6682)
                                                                             at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1118)
                                                                             at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2619)
                                                                             at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2727) 
                                                                             at android.app.ActivityThread.-wrap12(ActivityThread.java) 
                                                                             at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1478) 
                                                                             at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                                             at android.os.Looper.loop(Looper.java:154) 
                                                                             at android.app.ActivityThread.main(ActivityThread.java:6121) 
                                                                             at java.lang.reflect.Method.invoke(Native Method) 
                                                                             at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889) 
                                                                             at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)

--- a/app/src/main/java/org/mozilla/focus/fragment/ListPanelDialog.java
+++ b/app/src/main/java/org/mozilla/focus/fragment/ListPanelDialog.java
@@ -138,7 +138,6 @@ public class ListPanelDialog extends DialogFragment {
     }
 
     private void showPanelFragment(PanelFragment panelFragment) {
-        panelFragment.setTargetFragment(this, UNUSED_REQUEST_CODE);
         getChildFragmentManager().beginTransaction().replace(R.id.main_content, panelFragment).commit();
     }
 

--- a/app/src/main/java/org/mozilla/focus/fragment/PanelFragment.java
+++ b/app/src/main/java/org/mozilla/focus/fragment/PanelFragment.java
@@ -7,14 +7,14 @@ import android.support.v4.app.Fragment;
 public class PanelFragment extends Fragment {
 
     protected void closePanel() {
-        ((ListPanelDialog)getTargetFragment()).dismiss();
+        ((ListPanelDialog)getParentFragment()).dismiss();
     }
 
     @Override
     public void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        if( ! (getTargetFragment() instanceof ListPanelDialog) ) {
-            throw new RuntimeException("PanelFragments should have its Target Fragment set to an instance of ListPanelDialog");
+        if( ! (getParentFragment() instanceof ListPanelDialog) ) {
+            throw new RuntimeException("PanelFragments needs its parent to be an instance of ListPanelDialog");
         }
     }
 

